File: src/typescript/src/transport/transaction/incoming-transaction-factory.ts
import { IncomingMessage } from './message/incoming-message';
import { IPolyBus } from '../../i-poly-bus';
import { PolyBusBuilder } from '../../poly-bus-builder';
import { IncomingTransaction } from './incoming-transaction';

/**
 * A method for creating a new transaction for processing a request.
 * This should be used to integrate with external transaction systems to ensure message processing
 * is done within the context of a transaction.
 */
export type IncomingTransactionFactory = (builder: PolyBusBuilder, bus: IPolyBus, message: IncomingMessage) => Promise<IncomingTransaction>;

File: src/python/src/transport/transaction/incoming_transaction_factory.py
"""Incoming transaction factory for creating incoming transactions in the PolyBus Python implementation."""

from typing import Callable, Awaitable
from .incoming_transaction import IncomingTransaction
from .message.incoming_message import IncomingMessage

IncomingTransactionFactory = Callable[
    ['PolyBusBuilder', 'IPolyBus', 'IncomingMessage'],
    Awaitable['IncomingTransaction']
]
"""
A method for creating a new transaction for processing a request.
This should be used to integrate with external transaction systems to ensure message processing
is done within the context of a transaction.
"""
